unlock-app/src/middlewares/walletMiddleware.js

```diff
@@ -151,7 +151,10 @@ export default function walletMiddleware({ getState, dispatch }) {
 
   return function(next) {
     // Connect to the current provider
-    walletService.connect(getState().provider)
+    // We connect once the middleware has been initialized, using setTimout (warning: fragile?)
```

Comment:
This solution should be fine for now, and I don't believe it is fragile even though it feels shitty :)

When we implement #1653 it will move the connect call out of this place in the middleware anyways, as we will want to call it when it is needed instead. In other words, it will be triggered by the UI instead of by redux directly.